addons/account/models/company.py

```diff
@@ -735,6 +736,22 @@ def build_move_info(move):
 
         return results_by_journal
 
+    @api.model
+    def _with_locked_records(self, records):
+        """ To avoid sending the same records multiple times from different transactions,
+        we use this generic method to lock the records passed as parameter.
+
+        :param records: The records to lock.
+        """
+        try:
+            with self.env.cr.savepoint(flush=False):
```

Comment:
I'll look into it.

It's on `res.company` because it's not only called on `account.move`, but also on `pos.order` in `l10n_mx_edi_pos`, `l10n_xm_edi.document` in `l10n_mx_edi` (in https://github.com/odoo/enterprise/pull/65602)
And maybe in more place in the future.